src/mail/view/ConversationViewModel.ts

```diff
@@ -0,0 +1,257 @@
+import { ConversationEntry, ConversationEntryTypeRef, Mail, MailTypeRef } from "../../api/entities/tutanota/TypeRefs.js"
+import { MailViewerViewModel } from "./MailViewerViewModel.js"
+import { CreateMailViewerOptions } from "./MailViewer.js"
+import { elementIdPart, firstBiggerThanSecond, getElementId, haveSameId, isSameId, listIdPart } from "../../api/common/utils/EntityUtils.js"
+import { assertNotNull, findLast, findLastIndex, groupBy } from "@tutao/tutanota-utils"
+import { EntityClient } from "../../api/common/EntityClient.js"
+import { LoadingStateTracker } from "../../offline/LoadingState.js"
+import { EntityEventsListener, EntityUpdateData, EventController, isUpdateForTypeRef } from "../../api/main/EventController.js"
+import { ConversationType, OperationType } from "../../api/common/TutanotaConstants.js"
+import { NotFoundError } from "../../api/common/error/RestError.js"
+
+export type MailViewerViewModelFactory = (options: CreateMailViewerOptions) => MailViewerViewModel
+
+export type MailItem = { type: "mail"; viewModel: MailViewerViewModel; entryId: IdTuple }
+export type SubjectItem = { type: "subject"; subject: string; id: string; entryId: IdTuple }
+export type DeletedItem = { type: "deleted"; entryId: IdTuple }
+export type ConversationItem = MailItem | SubjectItem | DeletedItem
+
+export class ConversationViewModel {
+	private readonly _primaryViewModel: MailViewerViewModel
+	private loadingState = new LoadingStateTracker()
+	private loadingPromise: Promise<void> | null = null
+	private conversation: ConversationItem[] | null = null
```

Comment:
we should be able to just have [primarySubject, primaryMail] as an initial conversation right?

Reply:
I'm not sure what you mean, we do have them (again in `entries()`)

Reply:
was just wondering if the null initial value is necessary if we get our primary mail already loaded in the constructor, that would get rid of some asserts and null checks.

Reply:
This was initially to figure out if we are loading or not but we also have `loadingState` so we could change it.